hare/broker.go

```diff
@@ -59,18 +58,17 @@ func NewBroker(networkService NetworkService) *Broker {
 	p.Closer = NewCloser()
 	p.network = networkService
 	p.outbox = make(map[uint32]chan Message)
+	p.inbox = networkService.RegisterGossipProtocol(ProtoName)
 
 	return p
 }
 
 // Start listening to protocol messages and dispatch messages (non-blocking)
 func (broker *Broker) Start() error {
-	if broker.inbox != nil { // Start has been called at least twice
-		log.Error("Could not start instance")
-		return StartInstanceError(errors.New("instance already started"))
-	}
-
-	broker.inbox = broker.network.RegisterGossipProtocol(ProtoName)
+	//if broker.inbox != nil { // Start has been called at least twice
+	//	log.Error("Could not start instance")
+	//	return StartInstanceError(errors.New("instance already started"))
+	//}
```

Comment:
you can move registration back here and uncomment above code. the only rule is that protocols should register before `swarm` is started.